support/hoe/deprecated_gem.rb

```diff
@@ -4,7 +4,15 @@
 # plugin loading.)
 module Hoe::Deprecated_Gem # rubocop:disable Style/ClassAndModuleCamelCase
   def linked_spec(spec)
-    atm = YAML.load(YAML.dump(spec))
+    permitted_classes = [
+      Gem::Specification, Gem::Version, Gem::Requirement, Time
+    ]
+    atm = if RUBY_VERSION >= '2.6'
+            YAML.safe_load(YAML.dump(spec),
+                           permitted_classes: permitted_classes)
+          else
+            YAML.safe_load(YAML.dump(spec), permitted_classes)
+          end
```

Comment:
Will have a look at it.

Reply:
Hi! I started with this because I thought this is the easier one, but it raises some questions and I would like to hear your opinion on that.
IMHO the most elegant solution would be to use [Gem::SafeYAML.safe_load](https://github.com/rubygems/rubygems/blob/749c1fb941332aab67d4423cabedfa6281b72996/lib/rubygems/safe_yaml.rb#L30) in `Hoe::Deprecated_Gem`. It takes care of handling different psych versions, their different `safe_load` signatures and `permitted_classes` and `permitted_symbols` and it does not add an extra dependency. BUT this would need to require at least rubygems 3.0.0.beta3 to handle this properly and I don't know if you want this.
Another option could be to add a copy of `Gem::SafeYAML.safe_load` to `Hoe::Deprecated_Gem`.
Both options work fine but the same problem raises with `Hoe::Gemspec2` plugin, which also needs to be changed to make this work. So if rubygems version requirement is not an option I would suggest adding a `gemspec2_safe_load_gemspec` method to `Hoe::Gemspec2` and `deprecated_gem_safe_load_gemspec` to `Hoe::Deprecated_Gem`. Different names because as far as I understood the `Hoe` plugin system plugins will be included in `Hoe` and then it depends on the loading order which method will be used in case they have the same name.